### PR TITLE
docs: document plugin prefixed env vars

### PR DESCRIPTION
**Pull request by crenshaw-dev**
Given the positive response to [this post](https://cloud-native.slack.com/archives/C01TSERG0KZ/p1652451003177999) and the consensus of the Argo CD maintainers, we should go forward with environment variable prefixing in 2.4.

### CODE REVIEW

File: docs/user-guide/config-management-plugins.md [hunk 1]
```diff
@@ -67,8 +67,6 @@ spec:
     command: [sh, -c, 'echo "{\"kind\": \"ConfigMap\", \"apiVersion\": \"v1\", \"metadata\": { \"name\": \"$ARGOCD_APP_NAME\", \"namespace\": \"$ARGOCD_APP_NAMESPACE\", \"annotations\": {\"Foo\": \"$FOO\", \"KubeVersion\": \"$KUBE_VERSION\", \"KubeApiVersion\": \"$KUBE_API_VERSIONS\",\"Bar\": \"baz\"}}}"']
   discover:
     fileName: "./subdir/s*.yaml"
-  allowConcurrency: true
-  lockRepo: false
```
Comment (crenshaw-dev) [inline docs/user-guide/config-management-plugins.md:71]: These are not actually applicable to sidecar plugins.

File: docs/operator-manual/upgrading/2.3-2.4.md [hunk 2]
```diff
@@ -108,9 +116,39 @@ spec:
...
+
+### Confirm sidecar plugins have all necessary environment variables
+
+A bug in < 2.4 caused `init` and `generate` commands to receive environment variables from the main repo-server
+container, taking precedence over environment variables from the plugin's sidecar.
+
+Starting in 2.4, sidecar plugins will not receive environment variables from the main repo-server container. Make sure
+that any environment variables necessary for the sidecar plugin to function are set on the sidecar plugin.
+
+argocd-cm plugins will continue to receiver environment variables from the main repo-server container.
```
Comment (leoluz) [inline docs/operator-manual/upgrading/2.3-2.4.md:154]: s/receiver/receive

File: docs/user-guide/config-management-plugins.md [hunk 3]
```diff
@@ -183,6 +176,18 @@ spec:
...
+!!! note
+    The `discover.command` command only has access to the above environment starting with v2.4.
+
+> v2.4
+
+Before reaching the `init.command`, `generate.command`, and `discover.command` commands, Argo CD prefixes all 
+user-supplied environment variables (#3 above) with `ARGOCD_ENV_`. This prevents users from directly setting 
+potentially-sensitive environment variables.
+
+If your plugin was written before 2.3 and depends on user-supplied environment variables, then you will need to update
```
Comment (leoluz) [inline docs/user-guide/config-management-plugins.md:188]: just confirming.. before 2.3 or 2.4?

**Review by leoluz — APPROVED** [review]
LGTM

**Comment by crenshaw-dev** [comment]
Cherry-picked onto 2.4.

**Pull request merged**